src/main/java/org/gridsuite/modification/server/modifications/GeneratorScaling.java

```diff
@@ -0,0 +1,182 @@
+import org.gridsuite.modification.server.dto.ScalingVariationInfos;
+import org.gridsuite.modification.server.utils.ScalingUtils;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+import java.util.concurrent.atomic.AtomicReference;
+import java.util.stream.Collectors;
+
+import static org.gridsuite.modification.server.NetworkModificationException.Type.GENERATOR_SCALING_ERROR;
+
+/**
+ * @author Seddik Yengui <Seddik.yengui at rte-france.com>
+ */
+
+public class GeneratorScaling extends AbstractScaling {
+    private final boolean isIterative;
+
+    public GeneratorScaling(GeneratorScalingInfos generatorScalableInfos) {
+        super(generatorScalableInfos);
+        this.isIterative = generatorScalableInfos.isIterative();
+    }
+
+    @Override
+    public void applyStackingUpVariation(Network network,
+                                         List<IdentifiableAttributes> identifiableAttributes,
+                                         ScalingVariationInfos generatorScalingVariation) {
+        AtomicReference<Double> sum = new AtomicReference<>(0D);
+        Scalable stackingUpScalable = Scalable.stack(identifiableAttributes.stream()
+                .map(equipment -> getScalable(equipment.getId())).toArray(Scalable[]::new));
+        stackingUpScalable.scale(network, getAsked(generatorScalingVariation, sum));
+    }
+
+    @Override
+    public void applyVentilationVariation(Network network,
+                                          Reporter subReporter,
+                                          List<IdentifiableAttributes> identifiableAttributes,
+                                          ScalingVariationInfos generatorScalingVariation) {
+        AtomicReference<Double> sum = new AtomicReference<>(0D);
+        List<Float> percentages = new ArrayList<>();
+        List<Scalable> scalables = new ArrayList<>();
+        var distributionKeys = identifiableAttributes.stream()
+                .filter(equipment -> equipment.getDistributionKey() != null)
+                .mapToDouble(IdentifiableAttributes::getDistributionKey)
+                .sum();
+        if (distributionKeys == 0) {
+            String message = "This mode is available only for equipment with distribution key";
+            ScalingUtils.createReport(subReporter, GENERATOR_SCALING_ERROR.name(), message, TypedValue.ERROR_SEVERITY);
+            throw new NetworkModificationException(GENERATOR_SCALING_ERROR, message);
+        }
+
+        identifiableAttributes.forEach(equipment -> {
+            scalables.add(getScalable(equipment.getId()));
+            percentages.add((float) ((equipment.getDistributionKey() / distributionKeys) * 100));
+        });
+        Scalable ventilationScalable = Scalable.proportional(percentages, scalables, isIterative);
+        ventilationScalable.scale(network, getAsked(generatorScalingVariation, sum));
```

Comment:
sum value passed to getAsked method must be calculated like in other methods (applyRegularDistributionVariation, applyProportionalToPmaxVariation ...) , with summing up the generators targetP values